providers/provider-metrics/src/main/java/io/servicecomb/provider/metrics/MetricsController.java

```
@@ -0,0 +1,75 @@
+ * limitations under the License.
+ */
+
+package io.servicecomb.provider.metrics;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.servo.Metric;
+
+import io.servicecomb.foundation.metrics.MetricsServoRegistry;
+import io.servicecomb.provider.rest.common.RestSchema;
+
+@RestSchema(schemaId = "metricsService")
+@RequestMapping(path = "/metrics", produces = MediaType.APPLICATION_JSON)
+public class MetricsController {
+
+  private static final Logger log = LoggerFactory.getLogger(MetricsController.class);
+
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  private MetricsServoRegistry registry = null;
+
+  @Autowired
+  public MetricsController(MetricsServoRegistry registry) {
+    this.registry = registry;
+  }
+
+  @RequestMapping(path = "/", method = RequestMethod.GET)
+  public String metrics() {
+    List<List<Metric>> metrics = registry.getMemoryObserver().getObservations();
+    String response = "{}";
+    if (!metrics.isEmpty()) {
+      Map<String, String> values = new HashMap<>();
+      for (Metric metric : metrics.get(0)) {
+        String key = metric.getConfig().getName() +
+            (metric.getConfig().getTags().containsKey("statistic") ? "." + metric.getConfig().getTags()
+                .getValue("statistic") : "");
+        values.put(key, metric.getValue().toString());
+      }
+      try {
+        response = mapper.writeValueAsString(values);
+      } catch (JsonProcessingException e) {
+        log.error("failed parse metrics info");
```

You just eat the exception.

---

Fixed